### PR TITLE
fixed memory management of khmer_exception

### LINKED ISSUE #935: C++ exceptions are initialized with a temporary string
When parsing data files, errors result in exceptions constructed like this: khmer_file_exception(err.str().c_str())

khmer-exceptions only store a pointer to a C string: 
khmer_exception(const char * msg) : _msg(msg) { }
const char * _msg;

Since the string returned by str() is temporary, that results in dangling pointers.
Changing the _msg member to a std::string would solve the problem.

### PR DESCRIPTION
**Pull request by anotherthomas**
fixes #935

### CODE REVIEW

**Comment by anotherthomas** [comment]
- [x] Is it mergeable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by anotherthomas** [comment]
ready for review!

**Comment by mr-c** [comment]
Jenkins, okay to test

**Comment by anotherthomas** [comment]
updated the Changelog

**Comment by mr-c** [comment]
Jenkins, add to test list

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,8 @@
+2015-04-13  Thomas Fenzl  <thomas.fenzl@gmx.net>
+
+   * lib/{khmer_exception.hh,{counting,hashbits,hashtable,subset}.cc}: changed khmer_exception to use std::string to
```
Comment (mr-c) [inline ChangeLog]: max 80 chars :-)

**Comment by mr-c** [comment]
Otherwise, LGTM

**Pull request merged**